src/hooks/use-observable.d.ts

```diff
@@ -0,0 +1,3 @@
+import { Observable } from 'rxjs/Observable';
+export type RestrictArray<T> = T extends any[] ? T : []
+export declare function useObservable<State, Inputs>(observable: Observable<State>, initialState: State, inputs: RestrictArray<Inputs>): State;
```

Comment:
I dig that you're supplying TS signatures too? Have you seen the issue about adding TypeScript support to 🍉? Someone did much of the work and it just needs to be cleaned up a little and submitted as a pull request

Reply:
> I assume the correct author will be preserved??

yeah, absolutely!